versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/extension/NessieDbAdapter.java

```diff
@@ -22,4 +22,6 @@
 
 @Target({ElementType.FIELD, ElementType.PARAMETER})
 @Retention(RetentionPolicy.RUNTIME)
-public @interface NessieDbAdapter {}
+public @interface NessieDbAdapter {
+  String configMethod() default "";
```

Comment:
Could you add javadoc with expectations of the config (e.g. `static`, etc)?

Reply:
Suggestion: As an alternative we could use another annotation, e.g. `NessieDbAdapterConfigMethod`, to be used on the config method directly.... but current approach also LGTM... no preference, just an idea.

Reply:
A minor benefit of a method-level annotation is that one can instruct the IDE not to worry about "method not used" checks with those methods.

Reply:
Done. However, after the rebase I realized that removing all the `with*()` methods makes the approach impossible, because you can no longer "update" a configuration. Any idea?

Reply:
> method-level annotation

Doesn't work, because you can have different DatabaseAdapter instances that require different DatabaseAdapterConfigs.

Reply:
> However, after the rebase I realized that removing all the with*() methods makes the approach impossible, because you can no longer "update" a configuration.

The config method can still do something like `ImmutableTxDatabaseAdapterConfig.builder().from(configArg)...build()`. Will that work?